testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -886,17 +890,26 @@ def test_segmentgroup_str(self, u):
         sg = u.segments[:10]
         assert str(sg) == '<SegmentGroup [<Segment 4AKE>]>'
 
+    def test_atomgroupresidue_dtype(self, u):
+        try:
+            rg = list(u.atoms[100:130].residues)
+            assert len(rg) == 3
```

Comment:
normally one would let `pytest`/Python propagate the exception and just reference the issue in a comment `# see gh-2687`

Reply:
Only `rg = list(u.atoms[100:130].residues)` should be inside try/exept, everything else should be after it!

Reply:
The reason why I suggested this approach is because if we let the code raise ValueError then the test will fail differently that if it raised AssertionError. Because we know about one specific failure I suggested to make this known failure clear.

Reply:
Please see https://github.com/MDAnalysis/mdanalysis/pull/2693#discussion_r429423973 – I now concur with @tylerjereddy to _not_ use the try/except.